### ISSUE TITLE
Shortcut Keys not working on Joy pad.

### ISSUE THREAD

**Issue by nice-useles-person** [punesemu/puNES#67]
The shortcut cut keys are not working on Logitech F310 Joy Pad. The regular Game Pad Keys are working fine in both Xinput and Dinput Mode but shorcut keys are not working in any mode. I assign the shortcut keys of saving and loading state on left and right bumpers  respectively and they show up as assigned at the time but do not work at all and after closing and re-opening puNES, the keys assigned joy Pad Keys are reset to none automatically. I checked on both Xinput and Dinput seperately but it made no difference. Keyboard shortcuts are working fine as expected. Button Mapping for joy Pad keys for shortcuts seems broken. Kindly look into it...!

**Comment by nice-useles-person** [comment]
I checked in an older WIP .101, and it works fine in that version.

**Comment by punesemu** [comment]
Thx for the report. Could you try if you have the same problem with 0.102?

**Comment by nice-useles-person** [comment]
Checked the latest openGL version, same problem exist for OpenGL version as well. Let me check 0.102 now.

**Comment by nice-useles-person** [comment]
I just checked, Version .102 does not have this problem, but .103 and later versions with new GUI for settings have this problem as I just checked .103 as well as .104. Something probably broke with new interface was implemented.

**Comment by punesemu** [comment]
Ok, thx for the test. I will understand what the problem is and update you.

**Comment by punesemu** [comment]
Please, try this version [punes64_d3d9.exe](https://www.dropbox.com/s/iznqbvqlxpufjnd/punes64_d3d9.exe?dl=0) and tell me if works.

**Comment by nice-useles-person** [comment]
I started this version and set up shortcut keys for saving and loading states, then saved settings. After that I loaded a ROM in same session to test, but the keys were not working.
I closed puNES and Re-Opened to check if my shortcut keys were still assigned. They were shown assigned as I set them up before closing. Now, I loaded a ROM and they were working properly. So, I closed this test version.
Then, I opened the puNES version which previously had the problem, and checked if my keys were assigned there. They were shown as I assigned them in test version. So, I loaded a ROM to check them. Now, they were working in that version as well.
Then I assigned new keys in problem version to check if it can keep my assignment now. I closed and reopened the problem but the same keys were shown which were assigned by test version.

So, The shortcut keys only start working after restarting puNES and I think the new version with this problem are not saving the shortcut keys in input.cfg file so when we restart the puNES, shortcut keys are not loaded. But this test version saves the key configuration properly, so even the problematic version can load the proper settings.

In short, This version solved the problem under consideration. 👍

**Comment by punesemu** [comment]
> I started this version and set up shortcut keys for saving and loading states, then saved settings. After that I loaded a ROM in same session to test, but the keys were not working.

I'm interested in this. So, in the test version, as soon as you assign the shortcuts and start the rom, the new shortcuts did not work? If so, there is still some problem because the new settings should work immediately.

**Comment by nice-useles-person** [comment]
I deleted puNES folder in documents and then started puNES with clean settings. This time shortcut keys started working immediately after assigning them. So, You are right, they are working properly in test version. I do not know why they did not work immediately the first time.

**Comment by punesemu** [comment]
https://github.com/punesemu/puNES/commit/dd3389b365a49830cc4c4fbba84e11f0fefb4d2a and https://github.com/punesemu/puNES/commit/88d064fe5371729367fd70b60c87235d2972dc62. You can use the last WIP, it's better the the test version.
